packages/cozy-harvest-lib/src/services/budget-insight.js

```diff
@@ -166,10 +170,24 @@ export const createOrUpdateBIConnection = async ({
       }
     }
     logger.info('Creating or updating connection...')
-    const connection = await (connId
+    const isUpdate = Boolean(connId)
+    const connection = await (isUpdate
       ? updateBIConnection(config, connId, credsToSend, tempToken)
       : createBIConnection(config, credsToSend, tempToken))
 
+    if (!isUpdate) {
```

Comment:
nit: personnally I would have tested the throw would not happen during creation (in case)